### ISSUE TITLE
token → subtokens and merge subtokens

### ISSUE THREAD

**Issue by jonorthwash** [jonorthwash/ud-annotatrix#53]
Add functionality to the existing tokenisation routines (#2) so that tokens can be split into subtokens and adjacent subtokens can be merged.

**Comment by maryszmary** [comment]
Hm, not sure I understand. Can you give an example?

**Comment by jonorthwash** [comment]
I guess one example is "wanna".  After splitting this into two tokens, there should be a way to make them subtokens of a single token.

Another example is here:
```
1	Бұлардың	бұл	_	prn	dem|pl|gen	6	nmod:poss	_	_
2	бір	бір	_	num	_	3	nummod	_	_
3	ауыз	ауыз	_	n	nom	6	acl:relcl	_	_
4	бола	бол	_	v	iv|prc_impf	3	cop	_	_
5	алмаған	ал	_	vaux	neg|gpr_past	3	aux	_	_
6	себебі	себеп	_	n	px3sp|nom	7	nsubj	_	_
7-8	не	_	_	_	_	_	_	_	_
7	не	не	_	prn	itg|nom	0	root	_	_
8	_	е	_	cop	aor|p3|sg	7	cop	_	_
9	?	?	_	sent	_	7	punct	_	_
```

Let's say you want to make 7 and 8 separate tokens instead of subtokens.  There should be some way to do that in the interface.

**Comment by maryszmary** [comment]
Ah, I see, this is closely related to #8 and #36.

**Comment by jonorthwash** [comment]
Yes, those are both prerequisites to working on this, probably.  Those are the display- and format-level implementations; this is the editing-interface-level implementation, I guess.

**Comment by maryszmary** [comment]
If I got it right, now it works: left click on a token, then press `s`, then select with an arrow, which neighbor you want to merge with it:

1.
![image](https://user-images.githubusercontent.com/16348652/29295186-bbd267ee-815b-11e7-99a6-6213b4f479ce.png)

2.
![image](https://user-images.githubusercontent.com/16348652/29295193-c486736c-815b-11e7-94b5-07a07359d72a.png)

**Comment by jonorthwash** [comment]
Cool, what about splitting?

**Comment by maryszmary** [comment]
You mean, removing the supertoken?

**Comment by jonorthwash** [comment]
> You mean, removing the supertoken?

Yes, that's a good way to think about it.

**Comment by maryszmary** [comment]
Done.

1. Select the token to delete:
![image](https://user-images.githubusercontent.com/16348652/29366274-40d21c72-82a2-11e7-89a6-f40ea578a320.png)

2. Press `delete`:
![image](https://user-images.githubusercontent.com/16348652/29366286-502b3b68-82a2-11e7-99af-3201d479978e.png)

**Comment by jonorthwash** [comment]
Looking good!

**Comment by jonorthwash** [comment]
I'm having trouble splitting.  It says the feature isn't implemented?
![2017-08-28-15 26 23_001](https://user-images.githubusercontent.com/963849/29789851-64cfcb94-8c05-11e7-9105-4cd568658609.png)

**Comment by maryszmary** [comment]
It is not supported only for the sentences with spans because of the shifting issue. As I've written in #63, 

> The only thing affected by the index shift issue now is merging and splitting tokens, which turned out to be a bit trickier.

I'm working on it.

**Comment by jonorthwash** [comment]
Ah, I think I misunderstood you at the time.  Okay, cool.

**Comment by ftyers** [comment]
@maryszmary is this fixed now ? I see that #63 is fixed.
